src/libAtomVM/term.h

```diff
@@ -1080,4 +1080,33 @@ static inline int term_is_number(term t)
  */
 void term_display(FILE *fd, term t, const Context *ctx);
 
+/**
+ * @brief Checks if a term is a string (i.e., a list of characters)
+ *
+ * @details Returns 1 if a term is a list of characters or an empty list (nil term), otherwise 0.
+ * @param t the term that will be checked.
+ * @return 1 if check succedes, 0 otherwise.
+ */
+static inline int term_is_string(term t)
+{
+    if (term_is_list(t)) {
+        int proper;
+        term_list_length(t, &proper);
+        if (!proper) {
+            return 0;
+        }
+        term i = t;
+        while (term_is_nonempty_list(i)) {
+            term e = term_get_list_head(i);
+            if (!term_is_uint8(e)) {
+                return 0;
+            }
+            i = term_get_list_tail(i);
+        }
```

Comment:
We should check here if `i` is nil, if not this is not a proper list.

Reply:
See change above.  I think the change addresses this.